db-service/lib/deep-queries.js

```diff
@@ -246,7 +258,24 @@ const _getDeepQueries = (diff, target, root = false) => {
     if (op === 'create') {
       queries.push(INSERT.into(target).entries(diffEntry))
     } else if (op === 'delete') {
-      queries.push(DELETE.from(target).where(diffEntry))
+      // handle deletes differently to reduce the amount of sent queries
+
+      const keys = cds.utils
+        .Object_keys(target.keys)
+        .filter(key => !target.keys[key].virtual && !target.keys[key].isAssociation)
+
+      const keyVals = keys.map(k => ({ val: diffEntry[k] }))
+      const currDelete = deletes.get(target.name)
+      if (currDelete)
+        keys.length === 1
+          ? currDelete.DELETE.where[2].list.push(...keyVals)
+          : currDelete.DELETE.where[2].list.push({ list: keyVals })
+      else {
+        // if we have only one key, queries should be simpler
```

Comment:
I don't see the need for this simplification. Additionally this comes with an additional side effect for HANA where it will create a different execution plan if other parts of `CAP` don't add this simplification.

The final SQL different will be this for HANA:
 ```SQL
DELETE WHERE ID in JSON_TABLE(?,'$' COLUMNS(VAL STRING PATH '$.val'))

DELETE WHERE (ID) in JSON_TABLE(?,'$' COLUMNS(VAL1 STRING PATH '$.val1'))
```

So while I like the sentiment it might not achieve what it is trying to achieve. I would rather investigate what the actual behavior is before adding this additional logic.

Reply:
Yes, did the same as in lean draft impl, if the SQL queries are nice with the additional list, we can keep it like that also for single keys